### PR TITLE
Reconciler tests for channel deletion

### PR DESCRIPTION
**Pull request by aliok**
Fixes #

## Proposed Changes

- Reconciler tests for channel deletion
-
-

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

**Comment by pierDipi** [comment]
There are failed tests in GH actions
```
    --- FAIL: TestReconcileKind/Channel_spec_is_used_properly (0.18s)
        channel_test.go:1415: Extra update: &v1.Service{TypeMeta:v1.TypeMeta{Kind:"Service", APIVersion:"v1"}, ObjectMeta:v1.ObjectMeta{Name:"kc-kn-channel", GenerateName:"", Namespace:"test-nc", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), DeletionTimestamp:<nil>, DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string{"messaging.knative.dev/role":"kafka-channel"}, Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference{v1.OwnerReference{APIVersion:"messaging.knative.dev/v1beta1", Kind:"KafkaChannel", Name:"kc", UID:"c1234567-8901-2345-6789-123456789101", Controller:(*bool)(0xc0006558ff), BlockOwnerDeletion:(*bool)(0xc00065590f)}}, Finalizers:[]string(nil), ClusterName:"", ManagedFields:[]v1.ManagedFieldsEntry(nil)}, Spec:v1.ServiceSpec{Ports:[]v1.ServicePort(nil), Selector:map[string]string(nil), ClusterIP:"", ClusterIPs:[]string(nil), Type:"ExternalName", ExternalIPs:[]string(nil), SessionAffinity:"", LoadBalancerIP:"", LoadBalancerSourceRanges:[]string(nil), ExternalName:"kafka-channel-dispatcher.knative-testing.svc.cluster.local", ExternalTrafficPolicy:"", HealthCheckNodePort:0, PublishNotReadyAddresses:false, SessionAffinityConfig:(*v1.SessionAffinityConfig)(nil), IPFamilies:[]v1.IPFamily(nil), IPFamilyPolicy:(*v1.IPFamilyPolicyType)(nil), AllocateLoadBalancerNodePorts:(*bool)(nil), LoadBalancerClass:(*string)(nil), InternalTrafficPolicy:(*v1.ServiceInternalTrafficPolicyType)(nil)}, Status:v1.ServiceStatus{LoadBalancer:v1.LoadBalancerStatus{Ingress:[]v1.LoadBalancerIngress(nil)}, Conditions:[]v1.Condition(nil)}}
    --- FAIL: TestReconcileKind/Reconciled_normal_-_with_single_fresh_subscriber_-_with_auth_-_PlainText (0.61s)
        channel_test.go:1415: Extra update: &v1.Service{TypeMeta:v1.TypeMeta{Kind:"Service", APIVersion:"v1"}, ObjectMeta:v1.ObjectMeta{Name:"kc-kn-channel", GenerateName:"", Namespace:"test-nc", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), DeletionTimestamp:<nil>, DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string{"messaging.knative.dev/role":"kafka-channel"}, Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference{v1.OwnerReference{APIVersion:"messaging.knative.dev/v1beta1", Kind:"KafkaChannel", Name:"kc", UID:"c1234567-8901-2345-6789-123456789101", Controller:(*bool)(0xc0006543df), BlockOwnerDeletion:(*bool)(0xc0006543ef)}}, Finalizers:[]string(nil), ClusterName:"", ManagedFields:[]v1.ManagedFieldsEntry(nil)}, Spec:v1.ServiceSpec{Ports:[]v1.ServicePort(nil), Selector:map[string]string(nil), ClusterIP:"", ClusterIPs:[]string(nil), Type:"ExternalName", ExternalIPs:[]string(nil), SessionAffinity:"", LoadBalancerIP:"", LoadBalancerSourceRanges:[]string(nil), ExternalName:"kafka-channel-dispatcher.knative-testing.svc.cluster.local", ExternalTrafficPolicy:"", HealthCheckNodePort:0, PublishNotReadyAddresses:false, SessionAffinityConfig:(*v1.SessionAffinityConfig)(nil), IPFamilies:[]v1.IPFamily(nil), IPFamilyPolicy:(*v1.IPFamilyPolicyType)(nil), AllocateLoadBalancerNodePorts:(*bool)(nil), LoadBalancerClass:(*string)(nil), InternalTrafficPolicy:(*v1.ServiceInternalTrafficPolicyType)(nil)}, Status:v1.ServiceStatus{LoadBalancer:v1.LoadBalancerStatus{Ingress:[]v1.LoadBalancerIngress(nil)}, Conditions:[]v1.Condition(nil)}}
    --- FAIL: TestReconcileKind/Create_contract_configmap_when_it_does_not_exist (0.13s)
        channel_test.go:1415: Extra update: &v1.Service{TypeMeta:v1.TypeMeta{Kind:"Service", APIVersion:"v1"}, ObjectMeta:v1.ObjectMeta{Name:"kc-kn-channel", GenerateName:"", Namespace:"test-nc", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), DeletionTimestamp:<nil>, DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string{"messaging.knative.dev/role":"kafka-channel"}, Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference{v1.OwnerReference{APIVersion:"messaging.knative.dev/v1beta1", Kind:"KafkaChannel", Name:"kc", UID:"c1234567-8901-2345-6789-123456789101", Controller:(*bool)(0xc0008be49f), BlockOwnerDeletion:(*bool)(0xc0008be4bf)}}, Finalizers:[]string(nil), ClusterName:"", ManagedFields:[]v1.ManagedFieldsEntry(nil)}, Spec:v1.ServiceSpec{Ports:[]v1.ServicePort(nil), Selector:map[string]string(nil), ClusterIP:"", ClusterIPs:[]string(nil), Type:"ExternalName", ExternalIPs:[]string(nil), SessionAffinity:"", LoadBalancerIP:"", LoadBalancerSourceRanges:[]string(nil), ExternalName:"kafka-channel-dispatcher.knative-testing.svc.cluster.local", ExternalTrafficPolicy:"", HealthCheckNodePort:0, PublishNotReadyAddresses:false, SessionAffinityConfig:(*v1.SessionAffinityConfig)(nil), IPFamilies:[]v1.IPFamily(nil), IPFamilyPolicy:(*v1.IPFamilyPolicyType)(nil), AllocateLoadBalancerNodePorts:(*bool)(nil), LoadBalancerClass:(*string)(nil), InternalTrafficPolicy:(*v1.ServiceInternalTrafficPolicyType)(nil)}, Status:v1.ServiceStatus{LoadBalancer:v1.LoadBalancerStatus{Ingress:[]v1.LoadBalancerIngress(nil)}, Conditions:[]v1.Condition(nil)}}
    --- FAIL: TestReconcileKind/Do_not_create_contract_configmap_when_it_exists (0.16s)
        channel_test.go:1415: Extra update: &v1.Service{TypeMeta:v1.TypeMeta{Kind:"Service", APIVersion:"v1"}, ObjectMeta:v1.ObjectMeta{Name:"kc-kn-channel", GenerateName:"", Namespace:"test-nc", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), DeletionTimestamp:<nil>, DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string{"messaging.knative.dev/role":"kafka-channel"}, Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference{v1.OwnerReference{APIVersion:"messaging.knative.dev/v1beta1", Kind:"KafkaChannel", Name:"kc", UID:"c1234567-8901-2345-6789-123456789101", Controller:(*bool)(0xc00066c45f), BlockOwnerDeletion:(*bool)(0xc00066c46f)}}, Finalizers:[]string(nil), ClusterName:"", ManagedFields:[]v1.ManagedFieldsEntry(nil)}, Spec:v1.ServiceSpec{Ports:[]v1.ServicePort(nil), Selector:map[string]string(nil), ClusterIP:"", ClusterIPs:[]string(nil), Type:"ExternalName", ExternalIPs:[]string(nil), SessionAffinity:"", LoadBalancerIP:"", LoadBalancerSourceRanges:[]string(nil), ExternalName:"kafka-channel-dispatcher.knative-testing.svc.cluster.local", ExternalTrafficPolicy:"", HealthCheckNodePort:0, PublishNotReadyAddresses:false, SessionAffinityConfig:(*v1.SessionAffinityConfig)(nil), IPFamilies:[]v1.IPFamily(nil), IPFamilyPolicy:(*v1.IPFamilyPolicyType)(nil), AllocateLoadBalancerNodePorts:(*bool)(nil), LoadBalancerClass:(*string)(nil), InternalTrafficPolicy:(*v1.ServiceInternalTrafficPolicyType)(nil)}, Status:v1.ServiceStatus{LoadBalancer:v1.LoadBalancerStatus{Ingress:[]v1.LoadBalancerIngress(nil)}, Conditions:[]v1.Condition(nil)}}
FAIL
```

**Comment by aliok** [comment]
> There are failed tests in GH actions

That's actually because the system namespace is different in Prow and Github actions.

In Prow jobs it is set to `knative-eventing` and in Github actions it is set to `knative-testing`. 
I did set the env var in the tests now.

**Comment by aliok** [comment]
BTW, that failure was not related to my changes in the code. It was there I guess.

**Comment by aliok** [comment]
/test reconciler-tests_eventing-kafka-broker_main

Irrelevant I think

**Comment by pierDipi** [comment]
> BTW, that failure was not related to my changes in the code. It was there I guess.

Yes, there is something wrong with our prow unit tests set up, can you take a look?

File: control-plane/pkg/reconciler/channel/channel_test.go [hunk 1]
```diff
@@ -1349,6 +1350,71 @@ func TestReconcileKind(t *testing.T) {
...
+	t.Setenv("SYSTEM_NAMESPACE", "knative-eventing")
+
+	messagingv1beta.RegisterAlternateKafkaChannelConditionSet(base.IngressConditionSet)
+
+	env := *DefaultEnv
+	testKey := fmt.Sprintf("%s/%s", ChannelNamespace, ChannelName)
+
+	table := TableTest{
+		{
+			Name: "Finalize normal - no auth",
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/channel/channel_test.go:1364]: would you add more unit tests later or do you think only this is sufficient?

**Comment by pierDipi** [comment]
> /test reconciler-tests_eventing-kafka-broker_main
> 
> Irrelevant I think

By looking at the error, I guess we don't have retry configured ...

File: control-plane/pkg/reconciler/channel/channel_test.go (hunk 1 above)
Comment (aliok) [reply control-plane/pkg/reconciler/channel/channel_test.go:1364]: It is sufficient for now I think.

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Pull request merged**